### PR TITLE
Update pause image to v3.5

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Update dependencies and the test images to use pause 3.5. We also
provide a changelog entry for the new container image version.

#### Which issue(s) this PR fixes:

Refers to https://github.com/kubernetes/kubernetes/pull/97963

#### Special notes for your reviewer:

Previous 3.4.1 update PR: https://github.com/kubernetes/kubernetes/pull/98205

#### Does this PR introduce a user-facing change?

```release-note
Updated pause image to version 3.5, which now runs per default as pseudo user and group `65535:65535`.
This does not have any effect on remote container runtimes like CRI-O and containerd, which setup the pod
sandbox user and group on their own.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
None
```

### CODE REVIEW

**Review by troy0820 — CHANGES_REQUESTED** [review]
Change tag in file `build/pause/Makefile` to `3.5` to coincide with the changed dependency.

**Comment by saschagrunert** [comment]
We have to promote the new pause image, see https://github.com/kubernetes/k8s.io/pull/1793

**Comment by troy0820** [comment]
@saschagrunert Makes perfect sense.  Do we want to update that tag in a subsequent PR or after [1793](https://github.com/kubernetes/k8s.io/pull/1793) is merged in this PR?

**Comment by saschagrunert** [comment]
> @saschagrunert Makes perfect sense. Do we want to update that tag in a subsequent PR or after [1793](https://github.com/kubernetes/k8s.io/pull/1793) is merged in this PR?

`build/pause/Makefile` is already on `3.5` :)

/retest

**Comment by troy0820** [comment]
> build/pause/Makefile is already on 3.5 :)
🤦🏽  I was comparing locally
/lgtm

**Review by pacoxu — APPROVED** [review]
kubernetes/k8s.io#1793 is merged.
/lgtm

**Review by troy0820 — APPROVED** [review]

**Comment by pacoxu** [comment]
Like #98205
- containerd/kind/cri-o/minikube need update the pause image as well.

I opened the one for containerd. https://github.com/containerd/containerd/pull/5211

**Comment by troy0820** [comment]
I opened the one for KinD. https://github.com/kubernetes-sigs/kind/pull/2134

**Comment by neolit123** [comment]
hi, is this for 1.21?

**Comment by saschagrunert** [comment]
> hi, is this for 1.21?

There is no decision available yet, but I guess it would be good to have it in. :)

/retest

**Comment by saschagrunert** [comment]
I would need a couple of approvals on this one.

/assign @cblecker @janetkuo @sataqiu @yujuhong

**Comment by dims** [comment]
/assign @liggitt @smarterclayton 

easier to get root approvers :)

**Comment by liggitt** [comment]
just to double-check, this is not targeting 1.21, right?

**Comment by dims** [comment]
let's ask :)

@saschagrunert can you please chime in?

**Comment by saschagrunert** [comment]
> just to double-check, this is not targeting 1.21, right?

I see pros and cons of having it in 1.21: On one hand we have changed the pause image already and promoted it to k8s.gcr.io, which would make sense from my point of view to add this PR as well.

On the other hand we have to see how much work and CI related changes it will cause, which means I would definitely double check with @kubernetes/release-engineering and @kubernetes/sig-testing.

**Comment by liggitt** [comment]
> I see pros and cons of having it in 1.21: On one hand we have changed the pause image already and promoted it to k8s.gcr.io, which would make sense from my point of view to add this PR as well.

Even though a new pause image has been built and is available, it isn't actually in use yet. This seems like something that shouldn't be landing post code-freeze.

**Comment by justaugustus** [comment]
> This seems like something that shouldn't be landing during code-freeze.

Agreed.
/milestone v1.22

**Comment by BenTheElder** [comment]
Worth noting that containerd moved up to 3.5 already anyhow.

When dockershim goes we're not going to be able to control what version we use in CI from this repo at all.

I attempted to discuss this with SIG node but missed the meeting where this came up (conflicting hosting SIG testing at the same time), IMHO the forward path is moving pause to a k-sigs subproject owned by node and consumed by the CRI implementations on it's own release cycle (same as it having it's own versioning + promotion cycle today) and stop coupling it to k/k releases entirely.

**Comment by dims** [comment]
@BenTheElder the request in the sig-node meeting was to find OWNER(s) from sig-node community who can take charge of this effort, write down the rollout across the CRI impls and k8s, own the testing strategy, plan any other features that could be added etc.

**Comment by saschagrunert** [comment]
I still require approval by several people on this PR. Pinging @kubernetes/sig-cli-maintainers, @sig-node-approvers and @hasheddan (for SIG Release)

**Review by cpanato — APPROVED** [review]
/lgtm

**Review by hasheddan — APPROVED** [review]
/lgtm

File: build/pause/CHANGELOG.md
Comment (liggitt) [inline build/pause/CHANGELOG.md:5]: should this be 3.4.1?

Comment (saschagrunert) [reply build/pause/CHANGELOG.md]: Yep, fixed as suggested :+1:

**Comment by saschagrunert** [comment]
Fixed the changelog entry :heavy_check_mark:

**Comment by spiffxp** [comment]
As an end user I might find "pause runs as non-root user by default" more informative than "bumped to 3.5" as a release note. I see something like this is on a linked PR so maybe doesn't matter for this. However, does this require action on the part of someone upgrading their cluster if they for whatever reason require the previous behavior?

**Comment by BenTheElder** [comment]
> As an end user I might find "pause runs as non-root user by default" more informative than "bumped to 3.5" as a release note. 

Has anyone verified that this is actually the end result with any of the runtimes? How pause is used will depend on what runtime you use, e.g. upgrading it here typically only impacts dockershim to begin with.

> ... if they for whatever reason require the previous behavior?

... That shouldn't be possible? ™️

**Comment by saschagrunert** [comment]
> > As an end user I might find "pause runs as non-root user by default" more informative than "bumped to 3.5" as a release note.
> 
> Has anyone verified that this is actually the end result with any of the runtimes? How pause is used will depend on what runtime you use, e.g. upgrading it here typically only impacts dockershim to begin with.

CRI-O does not pickup the username for the sandbox from the pause container image. It still uses `0:0` as user:group.

**Comment by saschagrunert** [comment]
Had to rebase this one. Please take a look again @justaugustus @liggitt

**Comment by saschagrunert** [comment]
Had to rebase again.

**Comment by dims** [comment]
@saschagrunert we should probably add some release notes on what to expect / what to watch out for given the changes in https://github.com/kubernetes/kubernetes/pull/97963 

/approve

**Review by cpanato — APPROVED** [review]
/lgtm

**Comment by saschagrunert** [comment]
Updated  the changelog to be more descriptive. I also already addressed the changelog file, so:
/hold cancel

**Pull request merged**
